tt-train/CMakeLists.txt

```diff
@@ -0,0 +1,73 @@
+cmake_minimum_required(VERSION 3.18)
+cmake_policy(VERSION 3.18)
+include(cmake/compilers.cmake)
+
+if(DEFINED ENV{CMAKE_C_COMPILER} AND DEFINED ENV{CMAKE_CXX_COMPILER})
+    message(STATUS "Setting C and C++ compiler from environment variables")
+    set(CMAKE_C_COMPILER $ENV{CMAKE_C_COMPILER})
+    set(CMAKE_CXX_COMPILER $ENV{CMAKE_CXX_COMPILER})
+endif()
+
+if(CMAKE_CXX_COMPILER AND CMAKE_C_COMPILER)
+    message(STATUS "Using specifed C++ compiler: ${CMAKE_CXX_COMPILER}")
+    message(STATUS "Using specifed C compiler: ${CMAKE_C_COMPILER}")
+else()
+    message(STATUS "No C or C++ compiler specified, defaulting to Clang-17")
+    FIND_AND_SET_CLANG17()
+endif()
+
+project(ml-framework-cpp)
```

Comment:
Why is the project `ml-framework-cpp` but the directory is `tt-train`?  Which name should we use when referring to this (either in discussions or in code)?

Reply:
1st is the internal name. Second one is the new which we got 2-3 weeks ago. There is a chance it will be renamed again. So I am not sure if we want to replace generic name.